rust/examples/src/bin/simple_market_maker.rs

```diff
@@ -0,0 +1,179 @@
+    ticker: String,
+    /// Optionally include your keypair path. Defaults to your Solana CLI config file.
+    #[clap(short, long)]
+    keypair_path: Option<String>,
+}
+
+pub fn get_payer_keypair_from_path(path: &str) -> anyhow::Result<Keypair> {
+    read_keypair_file(&*shellexpand::tilde(path)).map_err(|e| anyhow!(e.to_string()))
+}
+
+pub fn get_network(network_str: &str) -> &str {
+    match network_str {
+        "devnet" | "dev" | "d" => "https://api.devnet.solana.com",
+        "mainnet" | "main" | "m" | "mainnet-beta" => "https://api.mainnet-beta.solana.com",
+        "localnet" | "localhost" | "l" | "local" => "http://localhost:8899",
+        _ => network_str,
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    // Connect to the solana network and get the market address
+    let args = Args::parse();
+    let config = match CONFIG_FILE.as_ref() {
+        Some(config_file) => Config::load(config_file).unwrap_or_else(|_| {
+            println!("Failed to load config file: {}", config_file);
+            Config::default()
+        }),
+        None => Config::default(),
+    };
+    let trader = get_payer_keypair_from_path(&args.keypair_path.unwrap_or(config.keypair_path))?;
+    let network_url = &get_network(&args.url.unwrap_or(config.json_rpc_url)).to_string();
+
+    let mut sdk = SDKClient::new(&trader, network_url).await?;
+    let market = args.market;
+    sdk.add_market(&market).await?;
+
+    let MarketMetadata {
+        base_mint,
+        quote_mint,
+        ..
+    } = sdk.get_market_metadata(&args.market);
+
+    loop {
+        let cancel_all_ix = sdk.get_cancel_all_ix(&market)?;
+        let mut ixs = vec![cancel_all_ix];
+
+        let response = reqwest::get(format!(
+            "https://api.coinbase.com/v2/prices/{}/spot",
+            args.ticker
+        ))
+        .await?
+        .json::<serde_json::Value>()
+        .await?;
+
+        // place a bid and ask at the fair price +/- edge
+        let fair_price = f64::from_str(response["data"]["amount"].as_str().unwrap())?;
+        let bid_price = fair_price * (1.0 - QUOTE_EDGE_BPS as f64 / 10000.0);
+        let ask_price = fair_price * (1.0 + QUOTE_EDGE_BPS as f64 / 10000.0);
+
+        if bid_price == 0.0 || ask_price == 0.0 {
+            println!("Bid or ask price is 0.0, skipping order placement, cancelling orders");
+            let txid = sdk.client.sign_send_instructions(ixs, vec![]).await?;
+            println!("Transaction sent: {}", txid);
+            continue;
+        }
+
+        let bid_size = QUOTE_SIZE_IN_QUOTE_UNITS / bid_price;
+        let ask_size = QUOTE_SIZE_IN_QUOTE_UNITS / ask_price;
+
+        let clock_account_data = sdk.client.get_account_data(&sysvar::clock::id()).await?;
+
+        let clock: Clock = bincode::deserialize(&clock_account_data)
+            .map_err(|_| anyhow::Error::msg("Error deserializing clock"))?;
+
+        let bid_order_packet = OrderPacket::Limit {
```

Comment:
why not use helper method for creating limit orders?

Reply:
Can change to using a helper. I wanted an example that explicitly states every parameter here

Reply:
let's put in both and leave one commented out, and both with heavy docs